Content.Server/VendingMachines/VendingMachineComponent.cs

```diff
@@ -20,6 +20,17 @@ public sealed class VendingMachineComponent : SharedVendingMachineComponent
 
         public bool Broken;
 
+        [DataField("ejectItemsOnHit")]
+        public bool EjectItemsOnHit = false;
+
+        [DataField("chanceToDispenseOnHit")]
+        public float DispenseChanceOnHit = 0.5f;
```

Comment:
Chance seems very high? Also at least make it a nullable float if it's going on the base component.